### PR TITLE
feat(server_family): Add backup/restore Prometheus metrics

### PR DESCRIPTION
**Pull request by BagritsevichStepan**
fixes dragonflydb#3210

### CODE REVIEW

File: src/server/server_family.cc [hunk 1]
```diff
@@ -1270,6 +1281,15 @@ void PrintPrometheusMetrics(const Metrics& m, DflyCmd* dfly_cmd, StringResponse*
   AppendMetricWithoutLabels("lua_blocked_total", "", m.lua_stats.blocked_cnt, MetricType::COUNTER,
                             &resp->body());
 
+  AppendMetricWithoutLabels("backup_count", "", m.loading_stats.backup_count, MetricType::COUNTER,
```
Comment (romange) [inline src/server/server_family.cc:1284]: naming convetion. ever increasing counters in open metrics must end with `_total`

File: src/server/server_family.cc [hunk 2]
```diff
@@ -880,6 +883,7 @@ void ServerFamily::LoadFromSnapshot() {
     if (std::error_code(load_path_result.error()) == std::errc::no_such_file_or_directory) {
       LOG(WARNING) << "Load snapshot: No snapshot found";
```
Comment (BagritsevichStepan) [inline src/server/server_family.cc:884]: Do we need to increase 'failed_restore_count' if no snapshot was found?

File: src/server/server_family.h [hunk 3]
```diff
@@ -350,6 +360,8 @@ class ServerFamily {
 
   mutable util::fb2::Mutex peak_stats_mu_;
   mutable PeakStats peak_stats_;
+
+  LoadingStats loading_stats_;
```
Comment (dranikpg) [inline src/server/server_family.h:364]: What's guarding it? 🙂 I assume we can access it in parallel from GetMetrics() and update it form the cron snapshot job

Comment (BagritsevichStepan) [reply src/server/server_family.h]: My mistake :cry: . Fixed it in the last commit

**Review by dranikpg — APPROVED** [review]
👍🏻

**Pull request merged**